nixos/modules/services/networking/tinydns.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  dataFile = pkgs.writeText "tinydns-data" config.services.tinydns.data;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+    services.tinydns = {
+      enable = mkOption {
+        default = false;
+        description = "Whether to run the tinydns dns server";
+      };
+
+      data = mkOption {
+        description = "The DNS data to serve, in the format described by tinydns-data(8)";
+      };
+
+      ip = mkOption {
+        default = "0.0.0.0";
+        description = "IP address on which to listen for connections";
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.tinydns.enable {
+    environment = {
+      systemPackages = [ pkgs.djbdns ];
+    };
+
+    users = {
+      extraGroups.tinydns = {
+        gid = config.ids.gids.tinydns;
+      };
+
+      extraUsers.tinydns = {
+        uid = config.ids.uids.tinydns;
+        description = "tinydns user";
+        group = "tinydns";
+      };
+    };
+
+    systemd.services.tinydns = {
+      description = "djbdns tinydns server";
+      wantedBy = [ "multi-user.target" ];
+      path = with pkgs; [ daemontools djbdns ];
+      preStart = ''
+        rm -rf /var/lib/tinydns;
```

Comment:
If the data is nuked on every startup, is there a benefit to allocating a static uid?

Reply:
I think not, but I didn't realize until I saw @Mic92's commits that it was even _possible_ to _not_ allocate a static uid.  :)  Will look into it.